cmd/restic/cmd_ls.go

```diff
@@ -59,19 +71,55 @@ func runLs(opts LsOptions, gopts GlobalOptions, args []string) error {
 		return err
 	}
 
+	// extract any specific directories to walk
+	var dirs []string
+	if len(args) > 1 {
+		dirs = args[1:]
+	}
+
 	ctx, cancel := context.WithCancel(gopts.ctx)
 	defer cancel()
-	for sn := range FindFilteredSnapshots(ctx, repo, opts.Host, opts.Tags, opts.Paths, args) {
-		Verbosef("snapshot %s of %v at %s):\n", sn.ID().Str(), sn.Paths, sn.Time)
+	for sn := range FindFilteredSnapshots(ctx, repo, opts.Host, opts.Tags, opts.Paths, args[:1]) {
+		dirsToPrint := opts.Paths
+		if len(dirs) > 0 {
+			dirsToPrint = dirs
+		}
+		Verbosef("snapshot %s of %v at %s):\n", sn.ID().Str(), dirsToPrint, sn.Time)
 
 		err := walker.Walk(ctx, repo, *sn.Tree, nil, func(nodepath string, node *restic.Node, err error) (bool, error) {
 			if err != nil {
 				return false, err
 			}
-
 			if node == nil {
 				return false, nil
 			}
+
+			// apply any directory filters
+			if len(dirs) > 0 {
+				var nodeDir string
+				if !opts.Recursive {
+					// only needed for exact directory match; i.e. no subfolders
+					nodeDir = filepath.Dir(nodepath)
```

Comment:
`nodpath` is always slash-separated, as it is created by `path.Join()` (not `filepath.Join`), see https://godoc.org/github.com/restic/restic/internal/walker#WalkFunc

So we need to use `path.Dir()` here to make it work properly on Windows.

Reply:
Sounds good.